ERCS/erc-7769.md

```diff
@@ -0,0 +1,546 @@
+---
+eip: 7769
+title: JSON-RPC API for ERC-4337
+description: JSON-RPC API methods for communication between smart contract account wallets and ERC-4337 bundlers
+author: Vitalik Buterin (@vbuterin), Yoav Weiss (@yoavw), Dror Tirosh (@drortirosh), Shahaf Nacson (@shahafn), Alex Forshtat (@forshtat)
+discussions-to: https://ethereum-magicians.org/t/erc-7769-json-rpc-for-erc-4337-account-abstraction/21126
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-08-23
+requires: 4337, 7562
+---
+
+## Abstract
+
+Defines new JSON-RPC API methods which enable [ERC-4337](./eip-4337) wallets to communicate with `UserOpeation` mempool
+nodes and bundlers, matching the functionality that exists for Ethereum transactions.
+
+Additionally, a set of `debug` JSON-RPC API methods is defined in order to facilitate development, testing and
+debugging issues with ERC-4337 implementations.
+
+## Motivation
+
+In ERC-4337, user transactions as defined in Ethereum are replaced with `UserOperation` objects, which contain all the
+information needed to perform the operations requested by the users.
+
+However, existing Ethereum JSON-RPC API methods are not suited to working with `UserOperation` objects.
+In order to facilitate the operation of the alternative `UserOperation` mempool it is important that all
+implementations of the ERC-4337 protocol have a standardized set of APIs that can be used interchangeably.
+
+## Specification
+
+### Definitions
+
+* **bundler**: a node exposing the APIs, in order to submit them to the network.
+  A bundler collects one or more UserOperations into a bundle and submits them together to
+  the `EntryPoint` in a single `handleOps` call.
+
+### RPC methods (eth namespace)
+
+#### eth_sendUserOperation
+
+The `eth_sendUserOperation` method submits a `UserOperation` object to the UserOperation mempool.
+The client MUST validate the `UserOperation`, and return a result accordingly.
+
+The result SHOULD be set to the `userOpHash` if and only if the request passed simulation and was accepted
+in the client's UserOperation pool.
+
+If the validation, simulation, or UserOperation pool inclusion fails,
+`userOpHash` SHOULD NOT be returned. Rather, the client SHOULD return the failure reason.
+
+##### Parameters:
+
+1. **UserOperation** a full user-operation struct.\
+  All fields MUST be set as hex values.\
+  Empty `bytes` block (e.g. empty `initCode`) MUST be set to `"0x"`\
+2. **factory** and **factoryData**\
+  Must provide either both of these parameters, or none.
+3. **paymaster**, **paymasterData**, **paymasterValidationGasLimit**, **paymasterPostOpGasLimit**\
+  Must provide either all of these parameters, or none.
+4. **entryPoint** the `EntryPoint` contract address the request should be sent through.\
+  This MUST be one of the entry points returned by the `supportedEntryPoints` RPC call.
```

Comment:
Whenever a proposal is specifying anything to do with JSON, I suggest that you seriously consider using [JSON Schema](https://json-schema.org/) to define and document valid JSON. It's machine tractable, has a ton of tooling, and is way more specific than english text.